### PR TITLE
Adding advanced sections

### PR DESCRIPTION
**Pull request by mariobodemann**
* Advanced Types explains how to use different class to retrieve Assets, Entries and Spaces
* Advanced Searching explains how to retrieve entries matching more complex requests

### CODE REVIEW

File: docs/tutorials/android/advanced-filtering-and-searching.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
+---
+page: :docsAdvancedFilteringAndSearchingAndroid
+---
+
+This section will describe advanced usage of the Android and Java API, in order to filter, sort and order responses. The
+upcoming samples are using the following setup: `space id = cfexampleapi` and `api token = b4c0n73n7fu1`.
+
+## Searching
+
+In order query for all items, containing the complete string `nyan` anywhere ... (No matter if in name, description, or
```
Comment (neonichu) [inline docs/tutorials/android/advanced-filtering-and-searching.md]: Might be worth being more specific than "anywhere" — `Text` and `Symbol` fields are indexed, but e.g. not `Object`

File: docs/tutorials/android/advanced-filtering-and-searching.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+~~~ java
+CDAArray found = client.fetch(CDAEntry.class)
+    .where("skip", "1")
+    .all();
+~~~ 
+
+## Include
+
+If you do not want the query to return only top level entries, and not its children(the default), you could use the
```
Comment (neonichu) [inline docs/tutorials/android/advanced-filtering-and-searching.md]: I think this description is a little bit confusing and might lead to the belief than only 0 or 1 are valid values. Should mention that the value of `include` is the number of hierarchy levels being returned and that the default is 1, but that it is also possible to return more levels.

File: docs/tutorials/android/advanced-types.md [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+          "en-US": "Larry Wall"
+        },
+        "name": {
+          "en-US": "An introduction to regular expressions. Volume VI"
+        },
+        "description": {
+          "en-US": "Now you have two problems."
+        }
+      },
+      …
```
Comment (realityking) [inline docs/tutorials/android/advanced-types.md]: Have you tested this? I'm fairly certain it breaks the syntax highlighter.

Comment (mariobodemann) [inline docs/tutorials/android/advanced-types.md]: @realityking I just tried it and it looks like 
advanced-types-remark

Which part did you mean to not work?

Comment (realityking) [inline docs/tutorials/android/advanced-types.md]: Interesting. I remember having a problem with middleman in the past where errors in the JSON would be bright red. Apparently not the case anymore :)

**Pull request merged**